reactive/kotlinx-coroutines-rx2/src/main/kotlin/kotlinx/coroutines/experimental/rx2/RxAwait.kt

```diff
@@ -40,6 +40,44 @@ public suspend fun CompletableSource.await(): Unit = suspendCancellableCoroutine
     })
 }
 
+// ------------------------ MaybeSource ------------------------
+
+/**
+ * Awaits for completion of the maybe without blocking a thread.
+ * Returns the resulting value, null if no value was produced or throws the corresponding exception if this
+ * maybe had produced error.
+ *
+ * This suspending function is cancellable.
+ * If the [Job] of the current coroutine is completed while this suspending function is waiting, this function
+ * immediately resumes with [CancellationException].
+ */
+public suspend fun <T> MaybeSource<T>.await(): T? = suspendCancellableCoroutine { cont ->
```

Comment:
You're completely right. I also added a test for `Maybe.empty().await() == null`.